apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -30,15 +31,50 @@ namespace Google.Cloud.Diagnostics.Common
     internal sealed class SimpleManagedTracer : IManagedTracer
     {
         /// <summary>
-        /// A class that represents a running trace span.
+        /// A class that represents a trace span.
         /// </summary>
-        private class Span : IDisposable
+        internal sealed class Span : ISpan
         {
+            internal TraceSpan TraceSpan { get; private set; }
+
             private readonly SimpleManagedTracer _tracer;
-            public Span(SimpleManagedTracer tracer) { _tracer = tracer; }
+
+            private double _ticks;
+
+            internal Span(SimpleManagedTracer tracer, TraceSpan traceSpan)
+            {
+                _tracer = GaxPreconditions.CheckNotNull(tracer, nameof(tracer));
+                TraceSpan = GaxPreconditions.CheckNotNull(traceSpan, nameof(traceSpan));
```

Comment:
I prefer to always have checks like this but can remove them if you feel strongly